src/Workspaces/Core/Portable/Workspace/AdditionalDocumentEventArgs.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    public class AdditionalDocumentEventArgs : EventArgs
+    {
+        public AdditionalDocument Document { get; }
+
+        public AdditionalDocumentEventArgs(AdditionalDocument document)
+        {
+            Contract.ThrowIfNull(document);
```

Comment:
Since this is a public API, wouldn't throwing `ArgumentNullException` be better?